chop/constraints.py

```diff
@@ -446,14 +446,111 @@ def prox(self, x, step_size=None):
             output[g] = utils.bmul(output[g], renorm)
 
         return output
-        
 
 
 class Box:
-
+    """Box constraint."""
     def __init__(self, a, b):
-        self.a = a 
+        """
```

Comment:
I think the common practice is to put the init docstring in the class docstring

Reply:
see for instance https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_ridge.py#L603